pkg/scheduler/framework/runtime/registry_test.go

```diff
@@ -56,3 +58,108 @@ func TestRegistry_Filter(t *testing.T) {
 		})
 	}
 }
+
+func TestRegistry_Register(t *testing.T) {
+	var r = make(Registry)
+
+	tests := []struct {
+		name              string
+		pluginFactoryName string
+		wantErr           bool
+		expectedPlugins   []string
+	}{
+		{
+			name:              "p1-1",
+			pluginFactoryName: "p1",
+			wantErr:           false,
+			expectedPlugins:   []string{"p1"},
+		},
+		{
+			name:              "p2-1",
+			pluginFactoryName: "p2",
+			wantErr:           false,
+			expectedPlugins:   []string{"p1", "p2"},
+		},
+		{
+			name:              "p1-2",
+			pluginFactoryName: "p1",
+			wantErr:           true,
+			expectedPlugins:   []string{"p1", "p2"},
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			err := r.Register(tt.pluginFactoryName, mockPluginFactory)
```

Comment:
I think each sub-test shouldn't be coupled with the other. We can initialize the registry in every sub-test.
Maybe we can orginze the tests by :
```golang
	tests := []struct {
		name              string
		initialPlugins    []string
		registeringPlugin string
		wantErr           bool
		expectedPlugins   []string
	}{
		{
			name:              "plugin registered to a empty registry",
			initialPlugins:    nil,
			registeringPlugin: "foo",
			wantErr:           false,
			expectedPlugins:   []string{"foo"},
		},
		{
			name:              "plugin registered to a none empty registry",
			initialPlugins:    []string{"foo"},
			registeringPlugin: "bar",
			wantErr:           false,
			expectedPlugins:   []string{"foo", "bar"},
		},
		{
			name:              "plugin already exists",
			initialPlugins:    []string{"foo", "bar"},
			registeringPlugin: "foo",
			wantErr:           true,
			expectedPlugins:   []string{"foo", "bar"},
		},
	}
```

Reply:
Good point